api/revoke_test.go

```diff
@@ -0,0 +1,221 @@
+package api
+
+import (
+	"bytes"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/pkg/errors"
+	"github.com/smallstep/assert"
+	"github.com/smallstep/certificates/authority"
+	"github.com/smallstep/certificates/logging"
+)
+
+func TestRevokeRequestValidate(t *testing.T) {
+	type test struct {
+		rr  *RevokeRequest
+		err *Error
+		rts authority.RevocationTypeSelector
+	}
+	tests := map[string]test{
```

Comment:
Not for now, but you can look into integrating [gotests](https://github.com/cweill/gotests) in your editor.